### DISCUSSION TITLE
GestureRecognizers for net6 and prior versions

### DISCUSSION THREAD

**Discussion by Guidemarcus** [unoplatform/uno#10872] [Q&A] [answered]
Hi!

I was wondering if GestureRecognizers was supported on uno and if so how can I use it. I am trying to convert a xamarin.forms app to an uno net6 project. In my xamarin.forms app I am using

```xaml
<StackLayout Orientation="Horizontal">

         <StackLayout.GestureRecognizers>
                   <TapGestureRecognizer Command="{Binding ShowAvailableEnvironmentsCommand}"/>
          </StackLayout.GestureRecognizers>

          <Label Text="{Binding ViewModel.AlternateEnvironment}" Style="{StaticResource listSelectorLabel}" FontSize="16" />

          <Label Text="{x:Static utils:MaterialIcons.ChevronDown}"
                      Style="{StaticResource smallSuffixGlyphLabel}"
                      FontSize="16"
                      VerticalTextAlignment="Center"
                      HorizontalOptions="End" />
</StackLayout>
```

Is there an easy translation for GestureRecognizers? 

FYI, I know StackLayout and Label are going to become StackPanel and TextBlock.

**Comment by dr1rrb** [comment, marked as answer]
You should look at the "Manipulation events" https://learn.microsoft.com/en-us/windows/apps/design/input/touch-interactions#manipulation-events

Basically you only have to set the `ManipulationMode=TranslateX` (or `ManipulationMode=TranslateX|TranslateInertia` to enable inertia) on the element you want to drag and then listen to `ManipulationStarted`, `ManipulationDelta` and `ManipulationCompleted` events.

**Reply by dr1rrb** [reply]
Also regarding your code and the `TapGestureRecognizer`, for the tap there is a `Tapped` event directly on `UIElement`.

**Reply by Guidemarcus** [reply]
Thanks a lot! I will look into that. I thought about using the Tapped event on the UIElement, but I was also wondering about Pan and Swipes. I will look into ManipulationModes.
